pkg/publisher/iface.go

```diff
@@ -0,0 +1,7 @@
+package publisher
+
+import "context"
+
+type Publisher interface {
+	Publish(ctx context.Context, topic string, data interface{}) error
```

Comment:
Should we use bellow interface for `Publisher`, so it will be more general. Decode/encode will be independent of publisher, it's caller responsibility to encode data before publishing it and it can be json, protobuf, or other encoding standard.

```go
type Publisher interface {
	Publish(ctx context.Context, topic string, data []byte) error
}
```

Reply:
Yeah, I agree. This code is written 3 days ago, with my assumption of using only JSON Encoder to encode. Now, we can use other encoders, especially after Protobuf shows its performance.

I will change the interface & source code according to your suggestion.